models/trigger_map_item.go

```diff
@@ -26,91 +28,56 @@ const (
 
 const defaultDraftPullRequestEnabled = true
 
+type TriggerItemType string
+
+const (
+	CodePushType    TriggerItemType = "push"
+	PullRequestType TriggerItemType = "pull_request"
+	TagPushType     TriggerItemType = "tag"
+)
+
 type TriggerMapItemModel struct {
-	// Trigger target
-	PipelineID string `json:"pipeline,omitempty" yaml:"pipeline,omitempty"`
-	WorkflowID string `json:"workflow,omitempty" yaml:"workflow,omitempty"`
-	// Commit push event criteria
-	PushBranch string `json:"push_branch,omitempty" yaml:"push_branch,omitempty"`
-	// Tag push event criteria
-	Tag string `json:"tag,omitempty" yaml:"tag,omitempty"`
-	// Pull Request event criteria
-	PullRequestSourceBranch string `json:"pull_request_source_branch,omitempty" yaml:"pull_request_source_branch,omitempty"`
-	PullRequestTargetBranch string `json:"pull_request_target_branch,omitempty" yaml:"pull_request_target_branch,omitempty"`
-	DraftPullRequestEnabled *bool  `json:"draft_pull_request_enabled,omitempty" yaml:"draft_pull_request_enabled,omitempty"`
-
-	// Deprecated
+	// Trigger Item shared properties
+	Type       TriggerItemType `json:"type,omitempty" yaml:"type,omitempty"`
+	Enabled    *bool           `json:"enabled,omitempty" yaml:"enabled,omitempty"`
```

Comment:
Why is it not simply `bool`? Isn't it a required field?

Reply:
`enabled` is true by default and default fields can be omitted from the bitrise.yml. But the default value for a boolean field is false, so the `omitempty` struct tag was omitting the relevant value (when the trigger item is disabled).